narwhals/_pandas_like/utils.py

```diff
@@ -468,116 +486,104 @@ def get_dtype_backend(dtype: Any, implementation: str) -> str:
-            return "Int8"
-        else:
-            return "int8"
-    if isinstance_or_issubclass(dtype, dtypes.UInt64):
-        if dtype_backend == "pyarrow-nullable":
-            return "UInt64[pyarrow]"
-        if dtype_backend == "pandas-nullable":
-            return "UInt64"
-        else:
-            return "uint64"
-    if isinstance_or_issubclass(dtype, dtypes.UInt32):
-        if dtype_backend == "pyarrow-nullable":
-            return "UInt32[pyarrow]"
-        if dtype_backend == "pandas-nullable":
-            return "UInt32"
-        else:
-            return "uint32"
-    if isinstance_or_issubclass(dtype, dtypes.UInt16):
-        if dtype_backend == "pyarrow-nullable":
-            return "UInt16[pyarrow]"
-        if dtype_backend == "pandas-nullable":
-            return "UInt16"
-        else:
-            return "uint16"
-    if isinstance_or_issubclass(dtype, dtypes.UInt8):
-        if dtype_backend == "pyarrow-nullable":
-            return "UInt8[pyarrow]"
-        if dtype_backend == "pandas-nullable":
-            return "UInt8"
-        else:
-            return "uint8"
-    if isinstance_or_issubclass(dtype, dtypes.String):
-        if dtype_backend == "pyarrow-nullable":
-            return "string[pyarrow]"
-        if dtype_backend == "pandas-nullable":
-            return "string"
-        else:
-            return str
-    if isinstance_or_issubclass(dtype, dtypes.Boolean):
-        if dtype_backend == "pyarrow-nullable":
-            return "boolean[pyarrow]"
-        if dtype_backend == "pandas-nullable":
-            return "boolean"
-        else:
-            return "bool"
-    if isinstance_or_issubclass(dtype, dtypes.Categorical):
-        # todo: is there no pyarrow-backed categorical?
-        # or at least, convert_dtypes(dtype_backend='pyarrow') doesn't
-        # convert to it?
-        return "category"
-    if isinstance_or_issubclass(dtype, dtypes.Datetime):
-        # todo: different time units and time zones
-        if dtype_backend == "pyarrow-nullable":
-            return "timestamp[ns][pyarrow]"
-        return "datetime64[ns]"
-    if isinstance_or_issubclass(dtype, dtypes.Duration):
-        # todo: different time units and time zones
-        if dtype_backend == "pyarrow-nullable":
-            return "duration[ns][pyarrow]"
-        return "timedelta64[ns]"
-    if isinstance_or_issubclass(dtype, dtypes.Date):
-        if dtype_backend == "pyarrow-nullable":
-            return "date32[pyarrow]"
-        msg = "Date dtype only supported for pyarrow-backed data types in pandas"
-        raise NotImplementedError(msg)
+
+    for dtype_type, dtype_to_backend_mapping in dtypes_mapping.items():
+        if not isinstance_or_issubclass(dtype, dtype_type):
+            continue
+
+        if dtype_backend not in dtype_to_backend_mapping:
+            msg = f"Type {dtype_type} not supported for backend {dtype_backend}"
+            raise NotImplementedError(msg)
```

Comment:
not sure we should raise here - we just want to preserve the dtype backend if possible

Reply:
this will return the dtype backend, this raise is for the case of Date that only exists in PyArrow and not Pandas or Numpy